packages/codegen-ui-react/lib/action.ts

```diff
@@ -0,0 +1,137 @@
+/*
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+import ts, { Statement, factory, ObjectLiteralExpression, Expression } from 'typescript';
+import {
+  InvalidInputError,
+  StudioComponent,
+  StudioComponentChild,
+  ActionStudioComponentEvent,
+  StudioComponentProperty,
+} from '@aws-amplify/codegen-ui';
+import {
+  isFixedPropertyWithValue,
+  isBoundProperty,
+  isConcatenatedProperty,
+  isConditionalProperty,
+  buildBindingExpression,
+  buildBindingWithDefaultExpression,
+  buildConcatExpression,
+  buildConditionalExpression,
+  buildFixedLiteralExpression,
+  isActionEvent,
+} from './react-component-render-helper';
+
+enum Action {
+  'Amplify.Navigate' = 'Amplify.Navigate',
+}
+
+export default Action;
+
+export const ActionNameMapping: Partial<Record<Action, string>> = {
+  [Action['Amplify.Navigate']]: 'useNavigateAction',
+};
+
+export function isAction(action: string): action is Action {
+  return Object.values(Action).includes(action as Action);
+}
+
+export function getActionName(action: string): string {
+  const actionName = ActionNameMapping[Action[action as Action]];
+  if (actionName === undefined) {
+    throw new InvalidInputError(`${action} is not a valid action.`);
+  }
+  return actionName;
+}
+
+export function getComponentActions(component: StudioComponent | StudioComponentChild): ActionStudioComponentEvent[] {
+  // TODO: add unique identifier to each action
+  if (component.events) {
```

Comment:
You'll miss child-only events as written. You should just treat as two separate conditionals, and create the empty array at the top.